src/transactions/OfferExchange.cpp

```diff
@@ -50,7 +50,8 @@ canSellAtMost(LedgerTxnHeader const& header, LedgerTxnEntry const& account,
         return std::max({getAvailableBalance(header, account), int64_t(0)});
     }
 
-    if (trustLine && trustLine.isAuthorized())
+    if (trustLine &&
+        (header.current().ledgerVersion >= 10 || trustLine.isAuthorized()))
```

Comment:
Is this true because in protocol 10 and above we delete offers when trust is revoked? If so, a comment would be welcome.

Reply:
now that I think of it, I am not sure this buys us much vs the defensive coding version `trustLine.isAuthorizedToMaintainLiabilities()`

Reply:
switched to `isAuthorizedToMaintainLiabilities`